kyaml/fn/framework/framework.go

```diff
@@ -170,3 +321,120 @@ func (rl *ResourceList) Filter(api kio.Filter) error {
 	}
 	return nil
 }
+
+// SortItems sorts the ResourceList.items by apiVersion, kind, namespace and name.
+func (rl *ResourceList) SortItems() {
+	sort.Sort(sortRNodeObjects(rl.Items))
+}
+
+type sortRNodeObjects []*yaml.RNode
+
+func (o sortRNodeObjects) Len() int      { return len(o) }
+func (o sortRNodeObjects) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
+func (o sortRNodeObjects) Less(i, j int) bool {
+	idi := o[i].GetResourceIdentifier()
+	idj := o[j].GetResourceIdentifier()
+	idStrI := fmt.Sprintf("%s %s %s %s", idi.GetAPIVersion(), idi.GetKind(), idi.GetNamespace(), idi.GetName())
+	idStrJ := fmt.Sprintf("%s %s %s %s", idj.GetAPIVersion(), idj.GetKind(), idj.GetNamespace(), idj.GetName())
```

Comment:
I've implemented a sort like this before, so I'm definitely in favour of the framework providing one!
* IIRC there are edge cases (e.g. involving fields or GV parts being empty) where the sort doesn't come out the way you'd intuitively want, and I vaguely remember switching to a more field-aware strategy as a result. @jeremyrickard does this sound familiar? 
* Should the heart of this go in `kio`'s filters collection? It already has `FormatFilter`, which is for sorting within objects--it would be neat to have this be a formal filter that exposes field sorting as an option for a one-stop solution to making your output stable. Framework could still have `func (rl *ResourceList) SortItems() {` to wrap it of course.